### PR TITLE
Allow to specify a slice in detector pattern in fitEDD

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

File: easistrain/EDD/io.py [hunk 1]
```diff
@@ -271,6 +271,7 @@ def read_detector_pattern(
     dataset: Union[str, int, None],
     scanNumber: Union[str, int],
     detector_name: str,
+    detectorSliceIndex: Optional[int] = None,
```
Comment (woutdenolf) [inline easistrain/EDD/io.py:274]:
```python
detectorSliceIndex: Union[int,tuple] = tuple(),
```

then you don't need the ternary conditional operator

File: tests/EDD/test_fit.py [hunk 2]
```diff
@@ -26,44 +29,56 @@ def generate_config(tmp_path: Path, test_data_path: Union[Path, str]) -> dict:
...
-    cfg = generate_config(tmp_path, test_data_path)
+    cfg = generate_config(tmp_path, test_data_path, scan_length)
     sample, dataset = cfg["sample"], cfg["dataset"]
     n_scan = cfg["scanNumber"]
     name_h, name_v = (
         cfg["nameHorizontalDetector"],
         cfg["nameVerticalDetector"],
     )
+
+    detector_dim = scan_length if scan_length is not None else 1
```
Comment (woutdenolf) [inline tests/EDD/test_fit.py:47]: You really love the ternary conditional operator.  I personally find them impossible to read.

**Review by woutdenolf — APPROVED** [review]
LGTM. But I think you may also need to slice the motor position (sx in the case if ihme19)

File: tests/EDD/test_fit.py (hunk 2 above)
Comment (loichuder) [reply tests/EDD/test_fit.py:47]: Well yeah, I find them easier to read than a branching if. I guess it is a matter a taste.

**Comment by loichuder** [comment]
> LGTM. But I think you may also need to slice the motor position (sx in the case if ihme19)

I added slicing for all `positionners`. Is it good or was it only for one ?

File: easistrain/EDD/fitEDD.py [hunk 3]
```diff
@@ -59,7 +66,7 @@ def fitEDD(
                     raise KeyError(
                         f"{positioner} was not found in input positioners! Possible values: {input_positioners.keys()}"
                     )
-                pos_data = input_positioners[positioner][()]
+                pos_data = input_positioners[positioner][detectorSliceIndex]
```
Comment (woutdenolf) [inline easistrain/EDD/fitEDD.py:69]: Almost. The positioners either have as many points as the detectors (i.e. they are motors used in the scan) or they are scalars. You'll need to handle both cases.

**Review by woutdenolf — CHANGES_REQUESTED** [review]

**Pull request merged**